miden-tx/src/executor/mod.rs

```diff
@@ -53,12 +53,26 @@ impl<D: DataStore> TransactionExecutor<D> {
     }
 
     /// Puts the [TransactionExecutor] into debug mode.
-    /// 
+    ///
     /// When transaction executor is in debug mode, all transaction-related code (note scripts,
     /// account code) will be compiled and executed in debug mode. This will ensure that all debug
     /// instructions present in the original source code are executed.
     pub fn with_debug_mode(mut self, in_debug_mode: bool) -> Self {
```

Comment:
Does it make sense to call `with_debug_mode(false)`? I mean is there a use case where I enable and disable this during lifetime?

Reply:
It probably won't happen this way - but it may still be convenient to have the mode passed in via the argument in cases when we want to set the debug mode based on some config variable. For example, something like:
```Rust
let in_debug_mode = /* read from some config */
let executor = TransactionExecutor::new(data_store).with_debug_mode(in_debug_mode);
```